File: src/main/java/ru/stnk/RestTestAPI/repository/QuoteBinFiveMinuteRepository.java
package ru.stnk.RestTestAPI.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.stnk.RestTestAPI.entity.coins.QuoteBinFiveMinute;

import java.util.List;

@Repository
public interface QuoteBinFiveMinuteRepository extends JpaRepository<QuoteBinFiveMinute, Long> {
    QuoteBinFiveMinute findBySymbolName(String symbolName);
    List<QuoteBinFiveMinute> findBySymbolNameOrderByTimestampDesc(String symbolName);
}

File: src/main/kotlin/ru/stnk/resttestapi/repository/QuoteBinFiveMinuteRepository.kt
package ru.stnk.resttestapi.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.stnk.resttestapi.entity.coins.QuoteBinFiveMinute

@Repository
interface QuoteBinFiveMinuteRepository : JpaRepository<QuoteBinFiveMinute, Long> {
    fun findBySymbolName(symbolName: String): QuoteBinFiveMinute?
    fun findBySymbolNameOrderByTimestampDesc(symbolName: String): List<QuoteBinFiveMinute>?
}
